src/xtgeo/grid3d/_grid_etc1.py

```diff
@@ -1420,16 +1434,18 @@ def get_gridquality_properties(self):
         fresults,
     )
 
-    grdprops = xtgeo.GridProperties()
+    grdprops = GridProperties()
 
     for num, name in qcnames.items():
-        prop = xtgeo.GridProperty(self, name=name)
-        dtype = np.float32
-        if num in qcdiscrete:
-            dtype = np.int32
-            prop.isdiscrete = True
-            prop.codes = {0: "None", 1: name}
-        prop.values = fresults[num, :].astype(dtype)
+        discrete = num in qcdiscrete
+        dtype = np.int32 if discrete else np.float32
+        prop = GridProperty(
+            self,
+            name=name,
+            discrete=discrete,
+            values=fresults[num, :].astype(dtype),
+            codes={0: "None", 1: name} if discrete else None,
+        )
```

Comment:
I changed this to pass values, discrete status and codes upon initialization of the GridProperty. Then I avoided mypy complaints regarding the values..